writable.go

```diff
@@ -18,6 +18,13 @@ func BytesWritable(b []byte) []byte {
 	return b[4:]
 }
 
+func PutBytesWritable(raw []byte) []byte {
```

Comment:
oh, so I guess the question I missed there was: should it just accept a byte slice and yolo the first X bytes, where X is whatever number of bytes it needs?